Sensory/src/main/java/com/google/android/sensory/HomeFragment.kt

```diff
@@ -81,6 +85,59 @@ class HomeFragment : Fragment() {
     dialog.show()
   }
 
+  private fun setupSyncUploadProgress() {
+    lifecycleScope.launch {
+      mainActivityViewModel.syncUploadState.collect {
+        when (it) {
+          is SyncUploadState.Started,
+          is SyncUploadState.InProgress -> showSyncBanner(it)
+          is SyncUploadState.Completed -> hideSyncBanner(it)
+          is SyncUploadState.Failed -> hideSyncBanner(it)
+          is SyncUploadState.NoOp -> {
+            println("No operation required. Sync is in progress")
+          }
+        }
+      }
+    }
+  }
+
+  private fun showSyncBanner(syncUploadState: SyncUploadState) {
+    with(binding.uploadLayout) {
+      if (linearLayoutUploadStatus.visibility != View.VISIBLE) {
+        // may add fade in animation here later
+        linearLayoutUploadStatus.visibility = View.VISIBLE
```

Comment:
I am assuming this is the case when we start the upload, in that case do we just show the animation? How about the percentage completion?

Reply:
Yeah there will be 0% progress shown.